sdk/eventhub/azure-eventhub/azure/eventhub/_client_base.py

```diff
@@ -52,38 +53,66 @@ def _parse_conn_str(conn_str, kwargs):
-                # An arbitrary, absurdly large number, since you can't renew.
-                shared_access_signature_expiry = int(time.time() * 2)
-    if not (all((endpoint, shared_access_key_name, shared_access_key)) or all((endpoint, shared_access_signature))):
+    check_case = kwargs.pop("check_case", False)    # type: Optional[bool]
+    conn_settings = core_parse_connection_string(conn_str, case_sensitive_keys=check_case)
+    if check_case:
+        shared_access_key = conn_settings.get("SharedAccessKey")
+        shared_access_key_name = conn_settings.get("SharedAccessKeyName")
+        endpoint = conn_settings.get("Endpoint")
+        entity_path = conn_settings.get("EntityPath")
+    
+        # non case sensitive check when parsing connection string for internal use
+        for key, value in conn_settings.items():
+            # only sas check is non case sensitive for both conn str properties and internal use
+            if key.lower() == "sharedaccesssignature":
+                shared_access_signature = value
+                try:
+                    # Expiry can be stored in the "se=<timestamp>" clause of the token. ('&'-separated key-value pairs)
+                    shared_access_signature_expiry = int(
+                        shared_access_signature.split("se=")[1].split("&")[0]   # type: ignore
+                    )
+                except (
+                    IndexError,
+                    TypeError,
+                    ValueError,
+                ):  # Fallback since technically expiry is optional.
+                    # An arbitrary, absurdly large number, since you can't renew.
+                    shared_access_signature_expiry = int(time.time() * 2)
+
+    if not check_case:
+        endpoint = conn_settings.get("endpoint") or conn_settings.get("hostname")
+        if endpoint:
+            endpoint = endpoint.rstrip("/")
+        shared_access_key_name = conn_settings.get("sharedaccesskeyname")
+        shared_access_key = conn_settings.get("sharedaccesskey")
+        entity_path = conn_settings.get('entitypath')
+        shared_access_signature = conn_settings.get("sharedaccesssignature")
+
+    entity = cast(str, eventhub_name or entity_path)
+
+    # check that endpoint is valid
+    if not endpoint:
+        raise ValueError("Connection string is either blank or malformed.")
+    parsed = urlparse(endpoint)
+    if not parsed.netloc:
+        raise ValueError("Invalid Endpoint on the Connection String.")
+    host = cast(str, parsed.netloc.strip())
+
+    if any([shared_access_key, shared_access_key_name]) and not all(
+        [shared_access_key, shared_access_key_name]
+    ):
         raise ValueError(
-            "Invalid connection string. Should be in the format: "
-            "Endpoint=sb://<FQDN>/;SharedAccessKeyName=<KeyName>;SharedAccessKey=<KeyValue>"
+            "Connection string must have both SharedAccessKeyName and SharedAccessKey."
```

Comment:
may I ask why we're change the information in the error here? -- the previous one has some value in telling the right format about the connection string.

Reply:
Yeah, so I was trying to conform with the [error message in SB](https://github.com/Azure/azure-sdk-for-python/blob/b682f616902ec1ab969de3b47f41ccfa5055adad/sdk/servicebus/azure-servicebus/azure/servicebus/_base_handler.py#L113).
My logic then was that the `Should be in format` error message seemed to be saying that the connection string must include `SharedAccessKeyName` and `SharedAccessKey` in any case, even though they are not required if `SharedAccessSignature` is included. This error message seemed more clear that if one is included, both should be included. I'll try to combine the two to provide more info.